packages/terminal/src/browser/terminal-frontend-contribution.ts

```diff
@@ -500,6 +500,7 @@ export class TerminalFrontendContribution implements TerminalService, CommandCon
         } else if (op.mode === 'reveal') {
             this.shell.revealWidget(widget.id);
         }
+        this.updateCurrentTerminal(widget);
```

Comment:
this line too. 
if we open a terminal widget, we consider it as the `current terminal`.

Reply:
please revert, current means it was focused, it can only happen if `activate` was passed and you should await here for activation to be finished that clients can be sure about it

Any other logic is bogus. The single source of truth of focus state is the application shell, everything should be derived from there.

Reply:
Got it. I will add another function to the `TerminalService` to keep track of the last `actived/revealed terminal`

Reply:
Why do you need to know last revealed?

Reply:
because i want the task to run in the terminal widget that "visibly" stays at the top, and the `TerminalFrontendContribution.currenTerminal` is not always the one.

For example, as a user you can set focus to TerminalA, and after that, TerminalB is revealed. In this case, B stays on top of A. While `TerminalFrontendContribution.currenTerminal` still points to A, I want to run the next task in B.